lib/base_service.js

```diff
@@ -91,6 +127,22 @@ class BaseService {
             this._logger = new Logger(config.logging);
             this._setUpDNSCaching();
 
+            // Catch unhandled rejections & log them. This relies on bluebird.
+            if (!this._unhandledRejectionHandlerAdded) {
+                this._unhandledRejectionHandlerAdded = true;
+                process.on('unhandledRejection', this._logUnhandledException);
+            }
+
+            // Similarly, log uncaught exceptions. Also, exit.
+            if (!this._uncaughtExceptionHandlerAdded) {
+                this._uncaughtExceptionHandlerAdded = true;
+                process.on('uncaughtException', this._uncaughtExceptionHandler);
+            }
+
+            if (this._logger.errorHandler && this._logger.on) {
```

Comment:
As I've said in the previous review, this does not belong here, its internal affears of the logger.

Reply:
Moved to logger.js